app/oxalis/security/CombinedAuthenticatorDAO.scala

```diff
@@ -0,0 +1,99 @@
+  */
+class CombinedAuthenticatorDAO extends AuthenticatorDAO[CombinedAuthenticator] with SecuredBaseDAO[CombinedAuthenticator]{
+  override def collectionName: String = "combinedAuthenticators"
+
+  implicit object FiniteDurationFormat extends Format[FiniteDuration] {
+    def reads(json: JsValue): JsResult[FiniteDuration] = LongReads.reads(json).map(_.seconds)
+    def writes(o: FiniteDuration): JsValue = LongWrites.writes(o.toSeconds)
+  }
+
+  implicit def optionFormat[T: Format]: Format[Option[T]] = new Format[Option[T]]{
+    override def reads(json: JsValue): JsResult[Option[T]] = json.validateOpt[T]
+
+    override def writes(o: Option[T]): JsValue = o match {
+      case Some(t) ⇒ implicitly[Writes[T]].writes(t)
+      case None ⇒ JsNull
+    }
+  }
+
+  override protected implicit def formatter: OFormat[CombinedAuthenticator] = JsonUtil.oFormat(Json.format[CombinedAuthenticator])
+
+  override def findOne[V](attribute: String, value: V)(implicit w: Writes[V], ctx: DBAccessContext): Fox[CombinedAuthenticator] = findOne(Json.obj(attribute -> w.writes(value)))
+
+  override def find(id: String): Future[Option[CombinedAuthenticator]] = findOne("id", id)(implicitly[Writes[String]],GlobalAccessContext).futureBox.map(box => box.toOption)
+
+  //adds the new token (and removes the old one)
+  override def add(authenticator: CombinedAuthenticator): Future[CombinedAuthenticator] = for{
+    maybeOldAuthenticator <- findByLoginInfo(authenticator.loginInfo)
+    _ <- insert(authenticator)(GlobalAccessContext).futureBox
+  }yield{
+    if(maybeOldAuthenticator.isDefined) remove(maybeOldAuthenticator.get.id)
+    authenticator
+  }
+
+  override def update(authenticator: CombinedAuthenticator): Future[CombinedAuthenticator] =
+    findAndModify(Json.obj("loginInfoParameter" -> LoginInfo(CredentialsProvider.ID, authenticator.loginInfo.providerKey)), Json.obj("$set" -> Json.obj(
+      "id" -> authenticator.id,
+      "lastUsedDateTime" -> authenticator.lastUsedDateTime,
+      "expirationDateTime" -> authenticator.expirationDateTime,
+      "idleTimeout" -> authenticator.idleTimeout)), returnNew = true)(GlobalAccessContext).futureBox.map(_.get)
+
+  override def remove(id: String): Future[Unit] = for{
+    _ <- remove("id", id)(implicitly[Writes[String]], GlobalAccessContext).futureBox
+  }yield{
+    ()
+  }
+
+  def findByLoginInfo(loginInfo: LoginInfo): Future[Option[CombinedAuthenticator]] =
+    findOne("loginInfoParameter", loginInfo)(implicitly[Writes[LoginInfo]],GlobalAccessContext).futureBox.map(box => box.toOption)
+
+  //this method was used for deleting all tokens of one user, but because a user is only allowed to have one token, it is not necessary. Should it be removed?
+  def removeByEmail(email: String):Fox[WriteResult] = {
+    val test = remove("loginInfoParameter", LoginInfo(CredentialsProvider.ID, email))(implicitly[Writes[LoginInfo]], GlobalAccessContext)
+    return test
+  }
+}
+
+object JsonUtil {
```

Comment:
I think we already have a JsonHelper Object somewhere in braingames-libs/util – maybe all of this stuff could be moved there

Reply:
If I want to add 'oFormat' and the 'optionFormat' to braingames-libs/util should I open a new PR for this?